src/WebAppResolver.ts

```diff
@@ -1,53 +1,100 @@
+                            options: {
+                                skipToken
+                            }
+                        });
 
-                    uiUtils.listAllIterator(client.webApps.list()).then((sites) => {
-                        for (const site of sites) {
-                            const siteName: string = nonNullProp(site, 'name');
-                            const count: number = (this.siteNameCounter.get(siteName) ?? 0) + 1;
+                        const record = response.data as Record<string, AppServiceQueryModel>;
+                        Object.values(record).forEach(data => {
+                            const count: number = (resolver.siteNameCounter.get(data.name) ?? 0) + 1;
+                            resolver.siteNameCounter.set(data.name, count);
+                            const model = {
+                                id: data.id,
+                                name: data.name,
+                                type: data.type,
+                                kind: data.kind,
+                                location: data.location,
+                                resourceGroup: data.resourceGroup,
+                                status: data.properties?.state ?? 'Unknown',
+                            } as AppServiceDataModel;
+                            resolver.siteCache.set(data.id.toLowerCase(), model);
+                        });
 
-                            this.siteNameCounter.set(siteName, count);
-                            this.siteCache.set(nonNullProp(site, 'id').toLowerCase(), site);
+                        const nextSkipToken = response?.skipToken;
+                        if (nextSkipToken) {
+                            await fetchApps(nextSkipToken);
+                        } else {
+                            resolver.loaded = true;
+                            return;
                         }
-                        resolve();
-                    })
-                        .catch((reason) => {
-                            reject(reason);
-                        });
-                });
+                    }
+
+                    return await fetchApps();
+                }
+
+                this.listWebAppsTask = fetchAllApps(subContext, this);
             }
 
-            await this.listWebAppsTask;
-            const site = this.siteCache.get(nonNullProp(resource, 'id').toLowerCase());
+            if (!this.loaded) {
+                await this.listWebAppsTask;
+            }
+            const siteModel = this.siteCache.get(nonNullProp(resource, 'id').toLowerCase());
 
             const groupBy: string | undefined = getGlobalSetting<string>('groupBy', 'azureResourceGroups');
-            const hasDuplicateSiteName: boolean = (this.siteNameCounter.get(nonNullValueAndProp(site, 'name')) ?? 1) > 1;
+            const hasDuplicateSiteName: boolean = (this.siteNameCounter.get(nonNullValueAndProp(siteModel, 'name')) ?? 1) > 1;
```

Comment:
I'm now a bit nervous to run `nonNullValueAndProp` on `siteModel` since we found in functions that the create scenario led to sometimes getting undefined for this.  Is there a way to change this to use an optional?